### PR TITLE
fix(@schematics/update): ignore npm 404 error

### PR DESCRIPTION
**Pull request by tiaguinho**
This PR will fix the issue https://github.com/angular/angular-cli/issues/10614

### CODE REVIEW

**Comment by tiaguinho** [comment]
What do you guys think?
cc: @hansl @clydin

**Comment by tiaguinho** [comment]
@alan-agius4 just did it!

**Comment by tiaguinho** [comment]
@clydin we can use the _workspace_ configuration to search local packages and filter ```_getAllDependencies```.
This way, only *npm* packages will be pass to ```getNpmPackageJson```.

**Comment by tiaguinho** [comment]
Changed!

cc: @alan-agius4 @clydin

File: packages/schematics/update/update/npm.ts [hunk 1]
```diff
@@ -133,7 +133,14 @@ export function getNpmPackageJson(
     },
   );
 
-  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(shareReplay());
+  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(
+    shareReplay(),
+    catchError(err => {
+      logger.warn(err);
+
+      return EMPTY;
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: use `empty()` instead of `EMPTY`.

File: packages/schematics/update/update/npm.ts [hunk 2]
```diff
@@ -133,7 +133,14 @@ export function getNpmPackageJson(
     },
   );
 
-  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(shareReplay());
+  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(
+    shareReplay(),
+    catchError(err => {
+      logger.warn(err);
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: Can you change this to `logger.warn(err.message || err)`?  The user doesn't need the full stack trace.

**Comment by tiaguinho** [comment]
Hi @clydin just made the changes you asked.
If I miss something, please let me know.

Thanks!

File: packages/schematics/update/update/npm.ts (hunk 1 above)
Comment (hansl) [reply packages/schematics/update/update/npm.ts]: Disregard, `EMPTY` is fine (see https://rxjs.dev/api/index/const/EMPTY)

File: packages/schematics/update/update/npm.ts [hunk 3]
```diff
@@ -133,7 +133,14 @@ export function getNpmPackageJson(
     },
   );
 
-  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(shareReplay());
+  const response = from<NpmRepositoryPackageJson>(resultPromise).pipe(
+    shareReplay(),
+    catchError(err => {
+      logger.warn(err.message || err);
+
+      return empty();
```
Comment (hansl) [inline packages/schematics/update/update/npm.ts]: Sorry, please use `EMPTY` (see https://rxjs.dev/api/index/const/EMPTY).

**Comment by hansl** [comment]
Additionally, could you add a `// TODO: find some way to test this`? Testing npm requests is hard, but we should have proper integration tests with this (in some future).

File: packages/schematics/update/update/npm.ts (hunk 3 above)
Comment (tiaguinho) [reply packages/schematics/update/update/npm.ts]: hahaha, no problem.

**Pull request merged**